lib/matplotlib/tight_layout.py

```diff
@@ -123,6 +123,9 @@ def auto_adjust_subplotpars(fig, renderer,
     for subplots, ax_bbox, (num1, num2) in zip(subplot_list,
                                                ax_bbox_list,
                                                num1num2_list):
+        if all([not ax.get_visible() for ax in subplots]):
```

Comment:
Yes, that's why you need both approaches. if a subset of the subplots are not visible, then the way the code is currently, the invisible axes still have `get_tightbbox` called on them, which would also fail. You need to protect that (my original approach), *and* protect from a `union()` call on an empty list (your current approach).